app/controllers/saml_idp_controller.rb

```diff
@@ -14,6 +14,7 @@ class SamlIdpController < ApplicationController
   before_action :validate_saml_logout_param, only: :logout
   before_action :store_sp_data, only: :auth
   before_action :authenticate_user!, except: [:metadata, :logout]
+  before_action :confirm_two_factor_setup, only: :auth
```

Comment:
Seems like there should be some way to avoid duplicating this list.  But I'm happy to land this change as is for now.